pkg/data/management/podsecuritypolicytemplate_data.go

```diff
@@ -94,9 +95,54 @@ func addDefaultPodSecurityPolicyTemplates(management *config.ManagementContext)
 				},
 				ReadOnlyRootFilesystem: false,
 			},
-			Description: "This is the default restricted Pod Security Policy Template.  It restricts many user " +
+			Description: "This is the default restricted Pod Security Policy Template. It restricts many user " +
 				"actions and does not allow privilege escalation.",
 		},
+		{
+			ObjectMeta: v12.ObjectMeta{
+				Name: "restricted-noroot",
+			},
+			Spec: v1beta1.PodSecurityPolicySpec{
+				Privileged:               false,
+				AllowPrivilegeEscalation: toBoolPtr(false),
+				RequiredDropCapabilities: []v1.Capability{
+					"ALL",
+				},
+				Volumes: []v1beta1.FSType{
+					"configMap",
+					"emptyDir",
+					"projected",
+					"secret",
+					"downwardAPI",
+					"csi",
```

Comment:
Couple of questions here:

1. The [upstream PSP](https://raw.githubusercontent.com/kubernetes/website/main/content/en/examples/policy/restricted-psp.yaml) includes access to `'ephemeral'` volumes. Would you mind elaborating on why this was excluded from the PSP that we include here?
2. The `restricted` PSP that we also include seems to not include the CSI volume. I may be missing the point of this PSP, but would we want to change this so that the only difference is the runAsUser rule?

Reply:
Answers:

1.  'ephemeral' volumes were not included in the new PSP, because it’s not supported in k8s 1.18. As soon as we dropped support for 1.18, we can add it.
```
# podsecuritypolicies.policy "restricted-noroot" was not valid:
# * spec.volumes: Unsupported value: "ephemeral": supported values: "*", "awsElasticBlockStore", 
"azureDisk", "azureFile", "cephFS", "cinder", "configMap", "csi", "downwardAPI", "emptyDir", "fc", 
"flexVolume", "flocker", "gcePersistentDisk", "gitRepo", "glusterfs", "hostPath", "iscsi", "nfs", 
"persistentVolumeClaim", "photonPersistentDisk", "portworxVolume", "projected", "quobyte", "rbd", 
"scaleIO", "secret", "storageos", "vsphereVolume"
```

2. The current `restricted` PSP that we include has some deviations from the `restricted` upstream, as you spot, the biggest one being:

```
				RunAsUser: v1beta1.RunAsUserStrategyOptions{
					Rule: runAsAny,
				},
```

Instead of `mustRunAsNonRoot`. We opted for creating a new "restricted" PSP in order to not break current deployment that are using the provided `restricted` PSP and to be as close as possible to the upstream policy.

Reply:
> is this just because we want to leave the current PSP alone and include any upstream changes in the restricted-noroot policy instead?

Yes, that's correct. We don't want to change the current provided `restricted` PSP. So all the changes were made to the new `restricted-noroot` PSP.